tests/detector/test_general.py

```diff
@@ -79,6 +79,55 @@ def test_schema_from_sparse_sample():
     }
 
 
+def test_schema_from_synthetic_sparse_sample():
+
+    # For each type (integer, number, string) there are example of
+    # the type ("is") and examples of other type ("not")
+    type_sample = {
+        "integer": {"is": 1, "not": "string"},
+        "number": {"is": 3.14, "not": "string"},
+        "string": {"is": "string", "not": 1},
+    }
+
+    # Columns with type and confidence
+    columns = [
+        {"type": "integer", "conf": 0.7},
+        {"type": "number", "conf": 1},
+        {"type": "string", "conf": 1},
+    ]
+
+    def generate_rows(num_rows=100, columns=[]):
+        rows = []
+        num_per_type = [num_rows * c["conf"] for c in columns]
+
+        for i in range(num_rows):
+            row = []
+            for ci, col in enumerate(columns):
+                if i < num_per_type[ci]:
+                    row.append(type_sample[col["type"]]["is"])
+                else:
+                    row.append(type_sample[col["type"]]["not"])
+
+            rows.append(row)
+
+        return rows
+
+    sample = generate_rows(columns=columns)
+    for conf in [0.6, 0.7, 0.8]:
```

Comment:
I think it's better to move it to `@pytest.mark.parametrize`

Reply:
Goot point, done!